wcfsetup/install/files/lib/system/user/authentication/password/algorithm/CryptMD5.class.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace wcf\system\user\authentication\password\algorithm;
+use wcf\system\user\authentication\password\IPasswordAlgorithm;
+
+/**
+ * Implementation of the password algorithm for MD5 mode of crypt().
+ *
+ * @author	Joshua Ruesweg
+ * @copyright	2001-2020 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package	WoltLabSuite\Core\System\User\Authentication\Password\Algorithm
+ * @since	5.4
+ */
+final class CryptMD5 implements IPasswordAlgorithm {
+	/**
+	 * @inheritDoc
+	 */
+	public function verify(string $password, string $hash): bool {
+		// The passwords are stored differently when importing. Sometimes they are saved with the salt,
+		// but sometimes also without the salt. We don't need the salt, because the salt is saved with the hash. 
+		$data = explode(':', $hash, 1);
+		
+		return \hash_equals($data[0], $this->hashWithSalt($password, $data[0]));
+	}
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function hash(string $password): string {
+		$salt = \bin2hex(\random_bytes(20));
```

Comment:
This salt does not work for `crypt()` and specifically not the MD5 based crypt.
```suggestion
		$salt = '$1$'.\bin2hex(\random_bytes(6)).'$'
```
should be acceptable.